Sources/NIOCore/FileHandle.swift

```diff
@@ -138,13 +142,15 @@ extension NIOFileHandle {
         public static let defaultPermissions = S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH
 #endif
 
+#if !os(WASI)
```

Comment:
wasi-libc has `mode_t` and `O_CREAT` but `O_CREAT` cannot be imported through ClangImporter for the same reason as `CLOCK_MONOTONIC`. Can we make a shim for it too?